### PR TITLE
Remove GuildController

### PR DESCRIPTION
**Pull request by MinnDevelopment**
[contributing]: https://github.com/DV8FromTheWorld/JDA/wiki/5%29-Contributing

## Pull Request Etiquette

- [x] I have checked the PRs for upcoming features/bug fixes.
- [x] I have read the [contributing guidelines][contributing].

### Changes

- [ ] Internal code
- [x] Library interface (affecting end-user code) 
- [ ] Documentation
- [ ] Other: \_____ 

Closes Issue: NaN

## Description

This moves all methods from `GuildController` into `Guild` and removes the old controller class.

### CODE REVIEW

**Comment by MinnDevelopment** [comment]
It might make sense to make some of these methods `default` in the `Guild` interface to reduce the size of the `GuildImpl` class.

**Comment by Andre601** [comment]
How would that affect the code?
Just remove `getController` from the code?

**Comment by binaryoverload** [comment]
This PR removes the guild controller and puts the methods on Guild instead @Andre601 so yes it does remove the `getController` call

**Comment by MinnDevelopment** [comment]
I added a method for discoverability used to kick members from a voice channel. `moveVoiceMember(member, null)` is now valid and can be used to kick a member from voice.

See discordapp/discord-api-docs#933

**Comment by MinnDevelopment** [comment]
Removing `removeRolesFromMember` and `addRolesToMember` is a difficult decision. The problem I've noticed over the years is that people don't even look for the `modifyMemberRoles` method and just intuitively do things like:

```java
c.addRolesToMember(member, mod).queue();
c.removeRolesFromMember(member, pleb).queue();
```

This seems to confuse a lot of users and they don't realize that these methods override the entire set of roles every time. Instead we will only promote atomic add/remove through `addRoleToMember` and `removeRoleFromMember` and the complete modify with `modifyMemberRoles`.
I hope this will reduce confusion in the future.

```java
Set<Role> add = Collections.singleton(mod);
Set<Role> remove = Collections.singleton(pleb);
c.modifyMemberRoles(member, add, remove).queue()
```

Naturally, there will still be people that don't use this correctly or just use add/remove in a loop but it is impossible to make this idiot-proof and that is also not the goal here.

File: src/main/java/net/dv8tion/jda/api/entities/Guild.java [hunk 1]
```diff
@@ -1709,6 +1700,1552 @@ default Emote getEmoteById(long id)
...
+     *         </ul>
+     * @throws net.dv8tion.jda.api.exceptions.HierarchyException
+     *         If attempting to set nickname for another member and the logged in account cannot manipulate the other user due to permission hierarchy position.
+     *         <br>See {@link net.dv8tion.jda.internal.utils.PermissionUtil#canInteract(Member, Member) PermissionUtil.canInteract(Member, Member)}
+     *
+     * @return {@link net.dv8tion.jda.api.requests.restaction.AuditableRestAction AuditableRestAction}
+     */
+    @Nonnull
+    @CheckReturnValue
+    AuditableRestAction<Void> setNickname(@Nonnull Member member, @Nullable String nickname);
```
[the remark this answers is not in the document]
Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/api/entities/Guild.java]: We have to consider what to do about other `setX` methods like `setMuted`

[the remark this answers is not in the document]
Comment (MinnDevelopment) [reply src/main/java/net/dv8tion/jda/api/entities/Guild.java]: I decided to rename `setMute` to `mute(member, bool)`

```java
void toggleMute(Member m) {
  Guild guild = member.getGuild();
  boolean muted = member.getVoiceState().isGuildMuted();
  guild.mute(member, !muted).queue();
}
```

And the same for `setDeafen`.

**Pull request merged**